chcemvediet/apps/obligees/tests/test_models.py

```diff
@@ -1,369 +1,672 @@
+                (u'Agency', u'Westside', u'agency', Obligee.STATUSES.PENDING),
+                ]
+        self.assertEqual([(h.name, h.street, h.slug, h.status) for h in history], expected_history)
+
+    def test_obligeealias_set_relation(self):
+        oblg = self._create_obligee(name=u'Agency')
+        obligeealias = self._create_obligeealias(obligee=oblg)
+        self.assertItemsEqual(oblg.obligeealias_set.all(), [obligeealias])
+
+    def test_obligeealias_set_relation_empty_by_default(self):
+        oblg = self._create_obligee()
+        self.assertItemsEqual(oblg.obligeealias_set.all(), [])
+
+    def test_branch_set_relation(self):
+        oblg = self._create_obligee()
+        branch = self._create_branch(obligee=oblg)
+        self.assertItemsEqual(oblg.branch_set.all(), [branch])
+
+    def test_branch_set_relation_empty_by_default(self):
+        oblg = self._create_obligee()
+        self.assertItemsEqual(oblg.branch_set.all(), [])
+
+    def test_inforequestdraft_set_relation(self):
+        oblg = self._create_obligee()
+        inforequestdraft = self._create_inforequest_draft(obligee=oblg)
+        self.assertItemsEqual(oblg.inforequestdraft_set.all(), [inforequestdraft])
+
+    def test_inforequestdraft_set_relation_empty_by_default(self):
+        oblg = self._create_obligee()
+        self.assertItemsEqual(oblg.inforequestdraft_set.all(), [])
+
+    def test_obligeetag_obligee_set_backward_relation(self):
+        obligeetag = self._create_obligeetag()
+        oblg1 = self._create_obligee(name=u'Obligee1')
+        oblg2 = self._create_obligee(name=u'Obligee2')
+        oblg1.tags.add(obligeetag)
+        oblg2.tags.add(obligeetag)
+        self.assertItemsEqual(obligeetag.obligee_set.all(), [oblg1, oblg2])
+
+    def test_obligeetag_obligee_set_backward_relation_empty_by_default(self):
+        obligeetag = self._create_obligeetag()
+        self.assertItemsEqual(obligeetag.obligee_set.all(), [])
+
+    def test_obligeegroup_obligee_set_backward_relation(self):
+        obligeegroup = self._create_obligeegroup()
+        oblg1 = self._create_obligee(name=u'Obligee1')
+        oblg2 = self._create_obligee(name=u'Obligee2')
+        oblg1.groups.add(obligeegroup)
+        oblg2.groups.add(obligeegroup)
+        self.assertItemsEqual(obligeegroup.obligee_set.all(), [oblg1, oblg2])
+
+    def test_obligeegroup_obligee_set_backward_relation_empty_by_default(self):
+        obligeegroup = self._create_obligeegroup()
+        self.assertItemsEqual(obligeegroup.obligee_set.all(), [])
+
     def test_no_default_ordering(self):
         self.assertFalse(Obligee.objects.all().ordered)
 
-    @unittest.skip(u'FIXME')
+    def test_dummy_email_property(self):
+        oblg = self._create_obligee(name=u'Agency')
+        self.assertEqual(oblg.dummy_email(oblg.name, u'{name}@chcemvediet.sk'), u'agency@chcemvediet.sk')
+
+    def test_dummy_email_property_with_accent_in_name(self):
```

Comment:
Pri testovanii interpunkcie a velkych/malych znakov si pre field `slug` pouzil samostatny test pre kazdy jeden pripad. Tu som to zlucil do jedneho testu. Moze to takto ostat alebo to mam tiez rozvetvit?

Reply:
@viliambalaz Kludne staci tak, ako si spravil.